assets/app/view/game/abilities.rb

```diff
@@ -94,6 +95,20 @@ def actions_for(company)
         @game.round.actions_for(company)
       end
 
+      # Render a button for the purchase train action that purchases the
+      # currently available depot train
+      def render_purchase_train_button
+        purchase = lambda do
+          process_action(Engine::Action::PurchaseTrain.new(
+            @selected_company
```

Comment:
Does it make sense to pass the train in as a parameter too, so that the action is more flexible for reuse?

Reply:
Sure, but can be added layer once needed.